pilot/pkg/networking/core/v1alpha3/cluster_builder_test.go

```diff
@@ -556,7 +586,27 @@ func TestApplyDestinationRule(t *testing.T) {
 					uint32(tt.destRule.TrafficPolicy.GetConnectionPool().GetHttp().MaxRequestsPerConnection) {
 					t.Errorf("Unexpected max_requests_per_connection found")
 				}
+			}
 
+			// Validate that alpn_override is configured on cluster correctly.
+			if tt.destRule != nil && tt.destRule.TrafficPolicy != nil {
+				md := tt.cluster.Metadata
+				istio := md.FilterMetadata[util.IstioMetadataKey]
+				if alpnOverride, found := istio.Fields["alpn_override"]; found {
+					if tt.destRule.TrafficPolicy.Tls.Mode == networking.ClientTLSSettings_SIMPLE {
+						if alpnOverride.GetStringValue() == "true" {
+							t.Errorf("alpn_override should be false for TLS mode SIMPLE")
+						} else if alpnOverride.GetStringValue() != "false" {
```

Comment:
This should just be an else; if for whatever reason another value is added (or alpn_override becomes a full-fleged enum field), we don't want to silently pass

Reply:
The following clause checks to see if it's a value other than true or false

Reply:
I reworked the if and removed the else entirely